documents/Design.md

```diff
@@ -0,0 +1,25 @@
+# Dataset Viewer Design
+
+Programming Language: Python 3  
+Coding Style: PEP8  
+Design Pattern: Model View Presenter (Passive View)
+## Class Diagram
+![Class Diagram](ClassDiagram.png)  
+DimensionView: Contains the slider, buttons, and stepper for an individual dimension.  
+DimensionPresenter: Manages the behavior of the DimensionView.  
+MainView: Contains the individual DimensionViews and the central plot.  
+MainPresenter: Manages the behavior of the MainView and retrieves data from the MainModel.  
+MainModel: Contains the data array. 
```

Comment:
More generally we will use nested MVP for ALL sub-components of the work. No UI component will be achieved outside of the MVP approach. Need to be mentioned in development principles.

Also, to prevent the main `Presenter` and `Model` becoming god objects, we should also include local models for our dimension MVP units.

Reply:
The development will process along the lines of a walking-skeleton comprising of vertical slices thorough the system layers. i.e. each pull request will deliver a feature that will correspond to MPV(s) and tests and include concrete realisations of classes required. In other words, the development will NOT attempt to fully tackle any single layer in one-go. i.e. it will not attempt to create all UIs and then wire-up the models and presenters.